### DISCUSSION TITLE
Problem with Print onload page

### DISCUSSION THREAD

**Discussion by jjhhrj** [Budibase/budibase#13354] [Help] [answered]
I'm new to budibase and I have a big trouble when trying to input a automation of printing.
![image](https://github.com/Budibase/budibase/assets/164453240/c30fe109-e175-4685-af4e-7e5877970e64)
I started by adding a automation app, then I coded in this js scripting
![image](https://github.com/Budibase/budibase/assets/164453240/7ffad0ef-2981-484b-8d74-cb9b7b377d9f)
but it didn't work as I wish although the automation was triggered.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @jjhhrj,

Automations are a backend feature.

The `window.print()` function is a JavaScript function that is typically used in frontend environments, such as web browsers, to open the print dialog box and allow the user to print the current page. It is part of the Web APIs provided by web browsers.

In a backend Node.js environment, there is no window object because Node.js runs JavaScript code outside of a browser context. Node.js is typically used for server-side scripting rather than client-side scripting. Therefore, functions like window.print() won't work in a Node.js environment because there is no browser to interact with.

If you need to generate printable content in a Node.js environment, you would typically use a library specifically designed for generating documents, such as pdfkit, puppeteer, or jsPDF, rather than relying on browser-specific functions like `window.print()`. These libraries allow you to programmatically generate PDF or other printable documents without relying on browser APIs.

Alas, this won't work as its something we would need to add on our end.

Hopefully this makes sense and is helpful for you.

**Reply by jjhhrj** [reply]
Do you mind help me to write the right code in this situation?
